### ISSUE TITLE
A way to reset a counter [core]

### ISSUE THREAD

**Issue by jaredmorgs** [asciidoctor/asciidoctor.org#550]
https://github.com/asciidoctor/asciidoctor/issues/1559 contains a docs enhancement concerning how `{counter:name:start}` is used in conjunction with the counter in docs.

*Next Actions:*

* review the content/suggestions in asciidoctor/asciidoctor#1559.
* incorporate the cited example into the guide.
* check whether there are other examples that may help users use counters to their advantage.

**Comment by mojavelinux** [comment]
The key information that is missing in this section is the relationship between the counter function and the document attribute. Specifically, the counter provides a way to increment the value of a document attribute. A regular attribute entry can be used to reset the value of the attribute. (In other words, the counter is just initializing and incrementing the value of a normal document attribute).

**Comment by rockyallen** [comment]
Now I am confused: from the above, I expect to get n=8, x=8 but the output is n=8, x=7

:n: 7
{counter:n}

:x: {n}

n={n}, x={x}

**Comment by mojavelinux** [comment]
Content for this section can be found in #561.

**Comment by mojavelinux** [comment]
The counter and the attribute aren't kept in sync (or it's sort of undefined behavior). An attribute entry should only be used for controlling a counter in the following ways:

* as an alternate way to declare an initial value
* to reset the value of the counter

In other words:

```
:n: 7

{counter:n} == 8
```

or

```
{counter:n:7} == 7

{counter:n} == 8

:!n:

{counter:n} == 1
```

More advance use simply breaks down because the counter mechanism was not designed to handle much more than that. If we want it to do more sophisticated things, then we'd need to fix the behavior in core.

**Comment by mojavelinux** [comment]
I think it's reasonable to add this disclaimer as a warning (or caution).

**Comment by mojavelinux** [comment]
This was incorporated into the fix for #561. See http://asciidoctor.org/docs/user-manual/#counters.
